pilot/pkg/kube/inject/inject.go

```diff
@@ -62,8 +63,10 @@ var (
 	}
 
 	annotationRegistry = map[string]annotationValidationFunc{
-		annotations.Register(annotationPolicy, "").Name:                                        alwaysValidFunc,
-		annotations.Register(annotationStatus, "").Name:                                        alwaysValidFunc,
+		annotations.Register(annotationPolicy, "").Name: alwaysValidFunc,
+		annotations.Register(annotationStatus, "").Name: alwaysValidFunc,
+		annotations.Register(annotationRewriteAppHTTPProbers,
+			"Rewrite HTTP readiness and liveness probes to be redirected to istio-proxy sidecar").Name: validateBool,
```

Comment:
Will this fail the inject? I would leave it not influence the normal injection, just ignore this annotation.

Reply:
Sorry @hzxuzhonghu , I don't get your point due to my lack understanding of the context.
Which injection will fail?
Do you mean to use `alwaysValidFunc` instead of `validateBool` ?

Reply:
yes, I mean if we set for example $annotationRewriteAppHTTPProbers: invalid-value, it will fail

Reply:
I think it's better to have it validated.
anyway, I will change to `alwaysValidFunc`
thanks